### PR TITLE
F mrbeams map

### PR DESCRIPTION
**Pull request by irlaec**

### CODE REVIEW

File: octoprint_mrbeam/static/js/laser_job_done_viewmodel.js [hunk 1]
```diff
@@ -26,6 +27,11 @@ $(function() {
 		});
 
         self.onStartupComplete = function(){
+            self.mapOptionEnabled = function () {
+                return self.settings.settings.plugins.mrbeam.dev.env().upper() === "PROD";
```
Comment (andywerner) [inline octoprint_mrbeam/static/js/laser_job_done_viewmodel.js]: Looks completely wrong. pls check

File: octoprint_mrbeam/static/js/laser_job_done_viewmodel.js [hunk 2]
```diff
@@ -73,15 +79,44 @@ $(function() {
...
+                doneDialog.removeClass('job-map-modal');
+                $('#job_done_info').show();
+                $('#share_job_and_location_btn').show();
+            }
+        };
+
+        self.map_btn = function(){
+            if (self.mapOptionEnabled) {
+                let duration = Math.floor(self.job_duration());
+                let cloud_function_trigger = "https://europe-west1-mrb-jobmap.cloudfunctions.net/generate_map?duration=" + duration + "&ts=" + Date.now();
```
Comment (andywerner) [inline octoprint_mrbeam/static/js/laser_job_done_viewmodel.js]: change name to "url_store_and_load_map" or so

File: octoprint_mrbeam/templates/laser_job_done.jinja2 [hunk 3]
```diff
@@ -1,3 +1,12 @@
+
+<link rel="stylesheet" href="https://unpkg.com/leaflet@1.4.0/dist/leaflet.css"
```
Comment (andywerner) [inline octoprint_mrbeam/templates/laser_job_done.jinja2]: put this in the html only if map is enabled, if env is DEV

File: octoprint_mrbeam/templates/laser_job_done.jinja2 [hunk 4]
```diff
@@ -6,15 +15,23 @@
             <h3 data-bind='visible: !is_dust_mode()'>{{ _('Laser Job Finished') }}</h3>
         </div>
         <div class="modal-body">
-            <img data-bind='visible: is_dust_mode()' src="/plugin/mrbeam/static/img/dusting.jpg">
-            <img data-bind='visible: !is_dust_mode()' src="/plugin/mrbeam/static/img/done.jpg">
-            <h5 data-bind='visible: is_dust_mode()'>Please wait a few seconds until Mr Beam II cleared most of the dust particles out of the working area.</h5>
-            <h5 data-bind='visible: !is_dust_mode()'>Done. Now check out what you have created with your Mr Beam II!</h5>
+            <!--Mr Beams map-->
+            <div id="mrbeams_map_container" class="z-depth-1-half map-container" style="height: 500px; width: 700px; display: none;">
```
Comment (andywerner) [inline octoprint_mrbeam/templates/laser_job_done.jinja2]: what happens if the according css is not loaded.

**Review by andywerner — CHANGES_REQUESTED** [review]

File: octoprint_mrbeam/static/js/laser_job_done_viewmodel.js (hunk 1 above)
Comment (irlaec) [reply octoprint_mrbeam/static/js/laser_job_done_viewmodel.js]: It was. Fixed!

File: octoprint_mrbeam/templates/laser_job_done.jinja2 (hunk 3 above)
Comment (irlaec) [reply octoprint_mrbeam/templates/laser_job_done.jinja2]: Done

File: octoprint_mrbeam/templates/laser_job_done.jinja2 (hunk 4 above)
Comment (irlaec) [reply octoprint_mrbeam/templates/laser_job_done.jinja2]: I've put this in an if statement, so it will only be shown if DEV mode

**Review by andywerner — APPROVED** [review]

**Pull request merged**